xonsh/completer.py

```diff
@@ -201,24 +210,29 @@ def find_and_complete(self, line, idx, ctx=None):
             endidx += 1
         endidx = endidx - 1 if line[endidx] in WS else endidx
         prefix = line[begidx:endidx+1]
-        return self.complete(prefix, line, begidx, endidx, ctx=ctx), begidx, endidx
+        return (self.complete(prefix, line, begidx, endidx, ctx=ctx),
+                begidx, endidx)
 
     def _add_env(self, paths, prefix):
+        _ = self
```

Comment:
It's a judgement call. If the method is not tightly coupled to this class then, yes, these functions should be moved to module scope. My change to ignore the unused `self` was the simplest change. But I'm happy to promote these methods to functions if that improves readability.